src/sections/AddressInfoSection.tsx

```diff
@@ -53,42 +53,61 @@ interface ParamTypes {
 
 const TransactionInfoSection = () => {
   const { id } = useParams<ParamTypes>()
-  const { client } = useGlobalContext()
-  const [addressInfo, setAddressInfo] = useState<AddressInfo>()
-  const [addressInfoError, setAddressInfoError] = useState('')
+  const { client, setSnackbarMessage } = useGlobalContext()
+  const [txNumber, setTxNumber] = useState<number>()
+  const [totalBalance, setTotalBalance] = useState<AddressBalance>()
   const [txListError, setTxListError] = useState('')
   const [txList, setTxList] = useState<Transaction[]>()
 
   const [infoLoading, setInfoLoading] = useState(true)
   const [txLoading, setTxLoading] = useState(true)
-  const previousId = useRef(id)
 
   // Default page
   const pageNumber = usePageNumber()
 
   // Address info
   useEffect(() => {
-    const fetchAddressInfo = async () => {
-      if (!client) return
-
-      previousId.current = id
+    if (!client) return
 
-      setInfoLoading(true)
+    const fetchTxNumber = async () => {
+      setTxNumber(undefined)
 
       try {
-        const response = await client.getAddressDetails(id)
-        if (!response.data) return
+        const { data } = await client.addresses.getAddressesAddressTotalTransactions(id)
+        setTxNumber(data)
+      } catch (error) {
+        console.error(error)
+        setSnackbarMessage({
+          text: getHumanReadableError(error, 'Error while fetching total transactions number'),
+          type: 'alert'
+        })
+      }
+
+      setInfoLoading(false)
```

Comment:
Should we have a more granular loading state (1 for each value)?

Reply:
YES I actually started working on it because I wanted to remove the redundant spinner next to the "Address" title (we have enough spinners I think), but I didn't want to delay the release :) Fixing!